src/Event/BeforeRegistrationStartsEvent.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\App\SDK\Event;
+
+use Psr\Http\Message\RequestInterface;
+use Shopware\App\SDK\Shop\ShopInterface;
+
+/**
+ * This event is fired before the registration process for new or already existing shops.
+ */
+class BeforeRegistrationStartsEvent extends AbstractAppLifecycleEvent
+{
+    public function __construct(
+        private readonly RequestInterface $request,
+        private readonly ShopInterface $shop,
+    ) {
+        parent::__construct($request, $shop);
+    }
+
+    public function getRequest(): RequestInterface
```

Comment:
Hm, sounds like all of functionality of this class has already implemented in [AbstractAppLifecycleEvent](https://github.com/shopware/app-php-sdk/blob/main/src/Event/AbstractAppLifecycleEvent.php).